helios-parser/src/main/kotlin/helios/parser/ByteBasedParser.kt

```diff
@@ -30,7 +30,7 @@ interface ByteBasedParser<J> : Parser<J> {
         var j = i
         var c: Int = (byte(j) and 0xff.toByte()).toInt()
         while (c != 34) {
-            if (c < 32) return die(j, "control char ($c) in string")
+            if (c < 32) die(j, "control char ($c) in string")
```

Comment:
This seems like a bug to me

Reply:
Unreachable code following IntelliJ because of `die` returns `Nothing`

Reply:
Literally an exception then